internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1672,8 +1672,35 @@ func ResetUserDisks(_ runtime.Sequence, data any) (runtime.TaskExecutionFunc, st
 	}, "resetUserDisks"
 }
 
+// ResetSystemDiskPartition represents the task to reset the system disk by target.
+func ResetSystemDiskPartition(target *installer.Target) func(seq runtime.Sequence, data any) (runtime.TaskExecutionFunc, string) {
+	return func(runtime.Sequence, any) (runtime.TaskExecutionFunc, string) {
+		return func(ctx context.Context, logger *log.Logger, r runtime.Runtime) (err error) {
+			bd := r.State().Machine().Disk().BlockDevice
+
+			pt, err := bd.PartitionTable()
+			if err != nil {
+				return fmt.Errorf("error reading partition table: %w", err)
+			}
+
+			_, err = target.Locate(pt)
+			if err != nil {
+				return fmt.Errorf("error locating partition with label %q: %w", target.Label, err)
+			}
+
+			if err = target.Format(); err != nil {
```

Comment:
Should we also drop and write partition here just like `ResetSystemDisk` does is Reset?

Reply:
noooo! :)